datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -497,6 +503,37 @@ def load_configuration_models(self, package_path=None):
             if instance_config is not None:
                 self._config_model_instance = instance_config
 
+    def load_memray_context_manager(self):
+        enable_memray = is_affirmative(self.instance.get('enable_memray', self.init_config.get('enable_memray', False)))
+
+        if enable_memray:
+            if PY2:
+                raise ConfigurationError("`enable_memray` option is not supported for py2 environments.")
+
+            if sys.platform not in ("linux", "darwin"):
+                raise ConfigurationError("`enable_memray` option is only supported on Linux and macOS.")
+
+            output_file = self.instance.get('memray_file', self.init_config.get('memray_file'))
+
+            # Append the check_id if the name if globally specified
+            if self.init_config.get('memray_file') and not self.instance.get('memray_file'):
+                output_file += ".{}".format(self.check_id)
+
+            native_mode = self.instance.get('memray_native_mode', self.init_config.get('memray_native_mode', False))
+
+            if output_file is None:
+                raise ConfigurationError('Setting `memray_file` must be provided to use memray.')
```

Comment:
For clarity, I'd at least move the `if output_file is None` to right after we define the `output_file` for the first time.

But this may be one of those cases where conditionals may be readable than all those `dict.get`s. If I read that right:

```suggestion
            if 'memray_file' in self.instance:
                output_file = self.instance['memray_file']
            elif 'memray_file' in self.init_config:
                output_file = "{}.{}".format(self.init_config['memray_file'], self.check_id)
            else:
                raise ConfigurationError('Setting `memray_file` must be provided to use memray.')

            native_mode = self.instance.get('memray_native_mode', self.init_config.get('memray_native_mode', False))
```